### ISSUE TITLE
New CoreDiagnostics Analysis issues

### ISSUE THREAD

**Issue by paulb777** [firebase/firebase-ios-sdk#3392]
Four new analysis issues showed up in last night's tests.

cc: @jasonhu-g 

Open Issues Introduced in This Integration
--
```
 
Analysis Issuein firebase-ios-sdk/GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTUploader.m:127Logic error: Branch condition evaluates to a garbage value (within a call to 'dispatch_sync') Analysis Issuein firebase-ios-sdk/GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTNanopbHelpers.m:170Coding conventions (Apple): Potential null dereference. According to coding standards in 'Creating and Returning NSError Objects' the parameter may be nullIntroduced by Michael Haney (783ee5b) Analysis Issuein firebase-ios-sdk/Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m:333Logic error: Converting a pointer value of type 'NSNumber *' to a primitive boolean value; instead, either compare the pointer to nil or call -boolValueIntroduced by Michael Haney (783ee5b) Analysis Issuein firebase-ios-sdk/Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m:394Logic error: Converting a pointer value of type 'NSNumber *' to a primitive boolean value; instead, either compare the pointer to nil or call -boolValueIntroduced by Michael Haney (783ee5b) |   | Analysis Issue |   | in firebase-ios-sdk/GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTUploader.m:127 |   | Logic error: Branch condition evaluates to a garbage value (within a call to 'dispatch_sync') |   |   |   |   | Analysis Issue |   | in firebase-ios-sdk/GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTNanopbHelpers.m:170 |   | Coding conventions (Apple): Potential null dereference. According to coding standards in 'Creating and Returning NSError Objects' the parameter may be null |   | Introduced by Michael Haney (783ee5b) |   |   | Analysis Issue |   | in firebase-ios-sdk/Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m:333 |   | Logic error: Converting a pointer value of type 'NSNumber *' to a primitive boolean value; instead, either compare the pointer to nil or call -boolValue |   | Introduced by Michael Haney (783ee5b) |   |   | Analysis Issue |   | in firebase-ios-sdk/Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m:394 |   | Logic error: Converting a pointer value of type 'NSNumber *' to a primitive boolean value; instead, either compare the pointer to nil or call -boolValue |   | Introduced by Michael Haney (783ee5b)
  | Analysis Issue
  | in firebase-ios-sdk/GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTUploader.m:127
  | Logic error: Branch condition evaluates to a garbage value (within a call to 'dispatch_sync')
  |  
 
  | Analysis Issue
  | in firebase-ios-sdk/GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTNanopbHelpers.m:170
  | Coding conventions (Apple): Potential null dereference. According to coding standards in 'Creating and Returning NSError Objects' the parameter may be null
  | Introduced by Michael Haney (783ee5b)
 
  | Analysis Issue
  | in firebase-ios-sdk/Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m:333
  | Logic error: Converting a pointer value of type 'NSNumber *' to a primitive boolean value; instead, either compare the pointer to nil or call -boolValue
  | Introduced by Michael Haney (783ee5b)
 
  | Analysis Issue
  | in firebase-ios-sdk/Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m:394
  | Logic error: Converting a pointer value of type 'NSNumber *' to a primitive boolean value; instead, either compare the pointer to nil or call -boolValue
  | Introduced by Michael Haney (783ee5b)

```

**Comment by mikehaney24** [comment]
These are already fixed in pre-release-6.2.0

**Comment by paulb777** [comment]
Will you merge into master to fix the Nightly Analysis runner?

**Comment by paulb777** [comment]
Fixed by #3396
